server/src/lib/model/achievements.ts

```diff
@@ -84,9 +85,9 @@ const bonus_condition_sql = {
     WHERE challenges.url_token = ?
     `,
   // Arguments: [client_id, client_id, challenge]
-  // Check if user has already won the bonus
+  // Check if user has already won the bonus, and if challenge is on-going now
   invite_contribute_same_session: `
-    SELECT earn.id IS NULL AS win_bonus, ? AS bonus_winner
+    SELECT (earn.id IS NULL) AND (NOW() BETWEEN start_date AND TIMESTAMPADD(WEEK, 3, start_date)) AS win_bonus, ? AS bonus_winner
```

Comment:
Same here - this should be available pre-challenge.

Reply:
Done